amoro-common/src/main/java/org/apache/amoro/resource/InternalResourceContainer.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.amoro.resource;
+
+/**
+ * Interface of internal resource container. For InternalResourceContainer, resources are managed by
+ * AMS. Resources are decoupled from processes. For ExternalResourceContainer, resources are managed
+ * outside, and each single resource belonged to a process.
+ */
+public interface InternalResourceContainer extends ResourceContainer {
```

Comment:
Do you mean the life cycle is managed by AMS?
Maybe remove the description about "external resource container"

Reply:
I've improved the description.  PTAL